lib/MetaCPAN/Model/Release.pm

```diff
@@ -271,6 +273,34 @@ sub _build_files {
     return \@files;
 }
 
+sub set_changes_file {
+    my $self  = shift;
+    my @files = @{ $_[0] };
+
+    my @changes_files = qw(
+        Changelog
+        ChangeLog
+        CHANGELOG
+        Changes
+        CHANGES
+        NEWS
+    );
+
+    if ( $files[0]->distribution eq 'perl' ) {
+        foreach my $file (@files) {
+            if ( $file->name eq 'perldelta.pod' ) {
+                return $file->path;
+            }
+
+        }
+    }
+
+    foreach my $file (@files) {
+        return $file->path if grep { $_ eq $file->name } @changes_files;
```

Comment:
We're checking against the basename... do we want to allow these files to exist anywhere other than the root?
It's unlikely for these names to be anything else (other than the changes file)
but it just feels like a bug waiting to happen.

I think we want to test against `$file->path`.  Does anyone think we _should_ check just `$file->name`?
If so, we may want to consider something like collect all the possible candidates and then choose the one with the shortest path.  As one example, this could easily match the change log for any dist (or other project) embedded in the release.  Also something like `CPAN::Changes` probably has several files like this to use for test data.

Either way, it would be good to have one of our fakecpan dists include a file matching one of these names in a subdirectory so that we are intentional about how it works.